apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/SessionPool.TargetedSessionPool.cs

```diff
@@ -487,9 +489,11 @@ public override Task<PooledSession> WithFreshTransactionOrNewAsync(PooledSession
             /// <summary>
             /// Release a session back to the pool (or refresh or evict it) and decrement the number of active sessions.
             /// </summary>
-            public override void Release(PooledSession session, bool deleteSession)
+            public override void Release(PooledSession session, ByteString transactionToRollback, bool deleteSession)
```

Comment:
Instead of receiving the transactionId here, which is always going to be the transaction associated with the session, I think I'd prefer passing the session to MaybeRollback... and checking there if the transaction has been commited or not and rollbacking otherwise. It's all in V1 so at most you have to make some properties internal instead of private. I think it'll be cleaner and we avoid the (minimal) risk of passing the wrong transaction to be rollbacked for a given session

Reply:
We don't actually have the transaction ID here in the case of PooledSession.ReleaseToPool - the session is already a fresh one without a transaction ID. We could change that of course, but I figured this was the minimally-disruptive change.

Reply:
Ah true, we have called AfterReset(). Yep it's fine then.